src/app/shared/filter/canned-range.component.ts

```diff
@@ -0,0 +1,320 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Subscription } from 'rxjs/Subscription';
+import { FilterModel } from '../../ngrx/model';
+import { castleFilter } from '../../ngrx/actions/castle.action.creator';
+import * as moment from 'moment';
+
+export const TODAY = 'Today';
+export const THIS_WEEK = 'This week';
+export const TWO_WEEKS = '2 weeks';
+export const THIS_MONTH = 'This month';
+export const THREE_MONTHS = '3 months';
+export const THIS_YEAR = 'This year';
+export const YESTERDAY = 'Yesterday';
+export const LAST_WEEK = 'Last week';
+export const PRIOR_TWO_WEEKS = 'Prior 2 weeks';
+export const LAST_MONTH = 'Last month';
+export const PRIOR_THREE_MONTHS = 'Prior 3 months';
+export const LAST_YEAR = 'Last year';
+
+@Component({
+  selector: 'metrics-canned-range',
+  template: `
+    <button class="btn-link" disabled="{{prevDisabled}}" (click)="prev()">&lt;&lt;&nbsp;PREV</button>
+    <div>
+      <span>When:</span>
+      <span>
+        <prx-select single="true" [options]="whenOptions" [selected]="selection" (onSelect)="onWhenChange($event)"></prx-select>
+      </span>
+    </div>
+    <button class="btn-link" disabled="{{nextDisabled}}" (click)="next()">NEXT&nbsp;&gt;&gt;</button>
+  `,
+  styleUrls: ['./canned-range.component.css']
+})
+export class CannedRangeComponent implements OnInit, OnDestroy {
+  filterStoreSub: Subscription;
+  filter: FilterModel;
+  whenOptions: any[];
+  selected: any[];
+  lastChosenRange: any[];
+
+  constructor(public store: Store<any>) {}
+
+  ngOnInit() {
+    this.filterStoreSub = this.store.select('filter').subscribe(state => {
+      this.filter = state;
+      this.genWhenDates();
+    });
+  }
+
+  ngOnDestroy() {
+    if (this.filterStoreSub) { this.filterStoreSub.unsubscribe(); }
+  }
+
+  endOfTodayUTC() {
+    return moment().utc().hours(23).minutes(59).seconds(59).milliseconds(999);
+  }
+
+  beginningOfTodayUTC() {
+    return moment().utc().hours(0).minutes(0).seconds(0).milliseconds(0);
+  }
+
+  genWhenDates() {
+    this.selected = null;
+
+    let utcEndDate = this.endOfTodayUTC();
+    let utcBeginDate = this.beginningOfTodayUTC();
+    const daysIntoWeek = utcEndDate.day();
+
+    this.whenOptions = [[TODAY, {beginDate: utcBeginDate.toDate(), endDate: utcEndDate.toDate()}]];
+    this.setSelectedIfFilterIsRange(this.whenOptions[this.whenOptions.length - 1]);
```

Comment:
could just iterate through all at end instead of calling `setSelected..` so many times? no strong feelings 🤷‍♀️ (wow, that emoji)

Reply:
done